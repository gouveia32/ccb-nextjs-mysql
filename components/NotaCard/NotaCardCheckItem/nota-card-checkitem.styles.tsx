import styled from "styled-components";

export const NotaCardCheckItemComponent = styled.div`
  width: 100%;
  display: flex;
  align-items: center;
`;

export const NotaCardCheckItemCheck = styled.div`
  //margin-right: 1rem;
`;

interface NotaCardCheckItemTextInterface {
  checked: boolean;
}

export const NotaCardCheckItemText = styled.span<
  NotaCardCheckItemTextInterface
>`
  flex: 1;
  display: flex;
  align-items: center;
  text-decoration: ${(props) => (props.checked ? "line-through" : "none")};
`;
