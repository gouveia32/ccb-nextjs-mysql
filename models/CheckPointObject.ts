export interface CheckPointInterface {
    id?: number | string;
    text: string;
    checked: boolean;
  }
  
  export type CheckPointType = CheckPointInterface;
  
  export const CheckPointObject: CheckPointType = {
           id: 0,
           text: "",
           checked: false,
         };
  
  export enum cCheckPoint {
    id = 'id',
    text = 'text',
    checked = 'checked',
  }