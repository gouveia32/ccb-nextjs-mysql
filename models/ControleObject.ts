export interface ControleInterface {
  id?: number | string;
  texto: string;
  marcado: boolean;
}

export type TipoControle = ControleInterface;

export const ControleObject: TipoControle = {
         id: 0,
         texto: "",
         marcado: false,
       };

export enum cControle {
  id = 'id',
  texto = 'texto',
  marcado = 'marcado',
}
