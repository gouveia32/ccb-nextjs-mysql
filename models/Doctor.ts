export interface DoctorModel {
  id: string;
  name: string;
  email: string;
  image?: string;
}

export type DoctorType = DoctorModel;

export const cDoctorModel = {
  id: "id",
  name: "name",
  email: "email",
  image: '',
};

export const DoctorObject: DoctorType = {
  id: "",
  name: "",
  email: "",
  image: '',
};

export function isDoctorType(object: any): object is DoctorType {
  return "name" in object;
}
