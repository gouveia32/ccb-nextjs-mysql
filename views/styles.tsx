import styled from "styled-components";
import { motion } from "framer-motion";
import { device } from "../resources/styles/utils/media-query-utils";

const dFlexCentered = `display: flex;
  justify-content: center;
  align-items: center;`;

export const LandingPageWrapper = styled(motion.section)`
  height: 100%;
  flex-direction: column;
  ${dFlexCentered}

  @media only screen and ${device.mobileL} {
    height: max-content;
    margin-left: -50px;
    margin-top: 2rem;
  }
`;

export const LandingPageContentWrapper = styled(motion.div)`
  ${dFlexCentered}
  
  @media only screen and ${device.mobileL} {
    flex-direction: column;
  }
`;

export const LandingPageCardWrapper = styled(motion.div)`
  flex-direction: column;
  ${dFlexCentered}

  @media only screen and ${device.mobileL} {
    margin: 0 !important;
  }
`;

export const LandingPageHeading = styled.h1`
  margin-bottom: 1rem;
  font-weight: bold;
  font-size: 4rem;
  color: #f5b500;
`;

export const LandingPageSubHeading = styled.h2`
  margin-top: 2rem;
  margin-bottom: 1rem;
  color: #8a8a8a;
`;

interface LandingPageBannerInterface {
  imageUrl?: string;
  color?: string;
  position?: string;
}

export const LandingPageBannerWrapper = styled.div`
  position: relative;
  height: 300px;
  width: 300px;
  overflow: hidden;
`;

export const LandingPageBanner = styled(motion.div)<LandingPageBannerInterface>`
  height: 300px;
  width: 300px;
  transform-origin: top;
  font-size: 2.8rem;
  font-family: "Roboto Slab", serif;
  font-weight: lighter;
  padding: 1rem;
  top: 0;
  background-size: cover;
  overflow: hidden;
  display: flex;
  flex-direction: column;

  position: ${(props) => (props.position ? props.position : "relative")};
  background: ${(props) =>
    props.imageUrl
      ? `url(${props.imageUrl})`
      : props.color
      ? props.color
      : "#ffe900"};
`;

export const LandingPageBannerText = styled(motion.h1)`
  font-family: "Shadows Into Light Two", cursive;
  font-size: 1.8rem;
  left: 0;
  padding: 10px;
  position: absolute;
  right: 0;
  text-align: center;
  top: 50%;
  -webkit-transform: translateY(-50%) rotate(-4deg);
  transform: translateY(-50%) rotate(-4deg);
`;
