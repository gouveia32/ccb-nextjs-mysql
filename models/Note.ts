import { TagType } from "./Tag";
import { CheckPointType } from "./CheckPointObject";
import { Note } from "@prisma/client";

export enum NoteTypeEnum {
  CHECK = "CHECK",
  TEXT = "TEXT",
}

export interface NoteModel {
  id?: string;
  name: string;
  noteType: NoteTypeEnum;
  content: string;
  color: string;
  pinned: boolean;
  createdAt: string;
  tags: TagType[];
  checkPoints?: CheckPointType[];
}

export type NoteType = NoteModel;

export const NoteObject: NoteType = {
  id: "",
  name: "",
  noteType: NoteTypeEnum.TEXT,
  content: "",
  color: "#fff",
  pinned: false,
  createdAt: new Date(Date.now()).toString(),
  tags: [],
  checkPoints: [],
};

export const NoteDBObject: Note = {
  id: "",
  name: "",
  noteType: "TEXT",
  color: "",
  content: "",
  pinned: false,
  createdAt: new Date(Date.now()),
  doctorId: "",
};

export enum cNoteModel {
  id = "id",
  name = "name",
  noteType = "noteType",
  color = "color",
  content = "content",
  pinned = "pinned",
  createdAt = "createdAt",
  tags = "tags",
  checkPoints = "checkPoints",
}
