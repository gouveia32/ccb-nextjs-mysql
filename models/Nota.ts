import { TipoPaciente } from "./Paciente";
import { TipoControle } from "./ControleObject";
import { Nota } from "@prisma/client";

export enum tipoNotaEnum {
  CHECK = "CHECK",
  TEXT = "TEXT",
}

export interface NotaModel {
  id?: string;
  nome: string;
  tipoNota: tipoNotaEnum;
  conteudo: string;
  cor: string;
  pino: boolean;
  criadoEm: string;
  pacientes: TipoPaciente[];
  controles?: TipoControle[];
}

export type TipoNota = NotaModel;

export const NotaObject: TipoNota = {
  id: "",
  nome: "",
  tipoNota: tipoNotaEnum.TEXT,
  conteudo: "",
  cor: "#fff",
  pino: false,
  criadoEm: new Date(Date.now()).toString(),
  pacientes: [],
  controles: [],
};

export const NotaDBObject: Nota = {
  id: "",
  nome: "",
  tipoNota: "TEXT",
  cor: "",
  conteudo: "",
  pino: false,
  criadoEm: new Date(Date.now()),
  medicoId: "",
};

export enum cNotaModel {
  id = "id",
  nome = "nome",
  tipoNota = "tipoNota",
  cor = "cor",
  conteudo = "conteudo",
  pino = "pino",
  criadoEm = "criadoEm",
  pacientes = "pacientes",
  controles = "controles",
}
