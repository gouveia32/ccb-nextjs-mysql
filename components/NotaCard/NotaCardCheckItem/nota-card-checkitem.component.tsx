import React from "react";
import {cCheckPoint, CheckPointType} from "../../../models/ControleObject";
import {Checkbox} from "@material-ui/core";
import {NotaCardCheckItemCheck, NotaCardCheckItemComponent, NotaCardCheckItemText,} from "./nota-card-checkitem.styles";

export interface NotaCardCheckItemProps {
  checkItem: CheckPointType;
  onChecked?: (checkItem: CheckPointType) => void;
}

const NotaCardCheckItem: React.FC<NotaCardCheckItemProps> = ({
  checkItem,
  onChecked,
}: NotaCardCheckItemProps) => {
  const handleChange = (attr: string, event: any) => {
    const newCheckItem: any = { ...checkItem };

    newCheckItem[attr] = event.target.checked;

    onChecked && onChecked(newCheckItem);
  };

  return (
    <NotaCardCheckItemComponent>
      <NotaCardCheckItemCheck>
        <Checkbox
          size={"small"}
          color={"default"}
          checked={checkItem.checked}
          className="p-1"
          onChange={(event) => handleChange(cCheckPoint.checked, event)}
          onClick={(event) => event.stopPropagation()}
        />
      </NotaCardCheckItemCheck>
      <NotaCardCheckItemText checked={checkItem.checked}>
        {checkItem.text}
      </NotaCardCheckItemText>
    </NotaCardCheckItemComponent>
  );
};

export default NotaCardCheckItem;
