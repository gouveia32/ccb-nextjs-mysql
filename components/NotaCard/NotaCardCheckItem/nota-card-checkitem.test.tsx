import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import { CheckPointObject } from "../../../models/ControleObject";
import { Checkbox } from "@material-ui/core";
import NotaCardCheckItem, {
  NotaCardCheckItemProps,
} from "./nota-card-checkitem.component";
import { NotaCardCheckItemText } from "./nota-card-checkitem.styles";

describe("NotaCardCheckItem component", () => {
  let wrapper: ShallowWrapper;
  let mockOnChecked: any;
  const checkPoint = CheckPointObject;
  checkPoint.checked = true;
  checkPoint.text = "text";

  beforeEach(() => {
    mockOnChecked = jest.fn();

    const mockProps: NotaCardCheckItemProps = {
      checkItem: checkPoint,
      onChecked: mockOnChecked,
    };

    wrapper = shallow(<NotaCardCheckItem {...mockProps} />);
  });

  it("should render NotaCardCheckItem component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call onChecked when Checkbox component is clicked", () => {
    wrapper.find(Checkbox).simulate("change", { target: { checked: true } });
    expect(mockOnChecked).toHaveBeenCalled();
  });

  it("should render when NotaCardCheckItemText component with checkItem values", () => {
    const checkItem = wrapper.find(NotaCardCheckItemText);
    expect(checkItem).toHaveLength(1);
    expect(checkItem.prop("checked")).toBe(checkPoint.checked);
    expect(checkItem.contains(checkPoint.text)).toEqual(true);
  });
});
