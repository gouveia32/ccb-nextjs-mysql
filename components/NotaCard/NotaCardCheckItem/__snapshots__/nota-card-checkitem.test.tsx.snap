// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteCardCheckItem component should render NoteCardCheckItem component 1`] = `ShallowWrapper {}`;
